src/v0/destinations/delighted/util.js

```diff
@@ -63,34 +62,35 @@ const userValidity = async (channel, Config, userId) => {
       },
       { destType: 'delighted', feature: 'transformation' },
     );
-    if (response && response.data && response.status === 200 && Array.isArray(response.data)) {
-      return response.data.length > 0;
-    }
-    throw new NetworkInstrumentationError('Invalid response');
-  } catch (error) {
-    let errMsg = '';
-    let errStatus = 400;
-    if (error.response && error.response.data) {
-      errMsg = JSON.stringify(error.response.data);
-      switch (error.response.status) {
-        case 422:
-        case 401:
-        case 406:
-        case 403:
-          errStatus = 400;
-          break;
-        case 500:
-        case 503:
-          errStatus = 500;
-          break;
-        default:
-          errStatus = 400;
+
+    if (!isHttpStatusSuccess(processedResponseDl.status)) {
+        let errMsg = '';
+        let errStatus = 400;
+        if (processedResponseDl.response) {
+          errMsg = JSON.stringify(processedResponseDl.response);
+          switch (processedResponseDl.status) {
+            case 422:
+            case 401:
+            case 406:
+            case 403:
+              errStatus = 400;
+              break;
+            case 500:
+            case 503:
+              errStatus = 500;
+              break;
+            default:
+              errStatus = 400;
+          }
```

Comment:
This logic seems to be handling network request for delighted APIs in general. What do you think about moving them to it's own `networkHandler.js` ?

Reply:
can take this up as part of separate PR after proper research